numpy/core/_methods.py

```diff
@@ -96,6 +97,10 @@ def _clip_dep_invoke_with_casting(ufunc, *args, out=None, casting=None, **kwargs
 def _clip(a, min=None, max=None, out=None, *, casting=None, **kwargs):
     if min is None and max is None:
         raise ValueError("One of max or min must be given")
+    elif min is not None and max is not None:
+        # need to handle scalar & array-like comparisons here
+        if (np.asanyarray(min) > np.asanyarray(max)).any():
+            raise ValueError("clip a_min greater than a_max")
```

Comment:
I think setting an `invalid` flag within the ufunc would make more sense here, the same way we do for `/ 0`

Reply:
Do we really want to make the ufunc substantially slower just for this sanity check? To me it makes more sense to define the behaviour and document it.

Reply:
I could go either way. We certainly don't want to invoke another ufunc in python, but the cost of another comparison might be marginal

Reply:
Within the ufunc, the cost is probably less, especially for the common case of constant min and max. 

On the other hand, with the present approach one can avoid the cost altogether by directly using the ufunc...

Overall, my sense remains to just be sure that the implementation matches the documentation, which states
```
    Equivalent to but faster than ``np.maximum(a_min, np.minimum(a, a_max))`
```
(which is consistent with the example of @tylerjereddy...), i.e., all we do is add a note that no check on `a_min <= a_max` is done.

Reply:
The [devdocs for `np.clip` ](https://www.numpy.org/devdocs/reference/generated/numpy.clip.html#numpy.clip) don't seem to have that statement, and that's what I wrote the property-based tests on, where the `Returns` entry indicates:

> An array with the elements of a, but where values < a_min are replaced with a_min, and those > a_max with a_max.

So, the proposal here is to amend the `numpy.clip` docstring with the pertinent content from `numpy.core.umath` in `ufunc_docstrings.py`, rather than take a performance hit & change behavior for the (admittedly bizarre) use case?

Reply:
Good point...  I guess for the devdocs, one would have to clarify the order in which the replacement is done (as written, it is not quite consistent with what you find).  But, yes, my suggestion is indeed to ensure the documentation is correct also for the corner case.

Reply:
Let's also check that I didn't accidentally swap the behavior from what it was before

Reply:
I went with "define and document" here for now, along with adjusting the cognate unit test to enforce the equivalence condition Marten dug up from the ufunc docs.

I'll mark this specific case as tentatively resolved to reduce review confusion.